### DISCUSSION TITLE
What function to call to get subscription notification messages?

### DISCUSSION THREAD

**Discussion by JeffMash** [amzn/selling-partner-api-models#3577] [General] [answered]
(Sorry for the newb question)

I created a destination and also a subscription for the ANY_OFFER_CHANGED notification.   It appears to be working because I can see messages when I go to Amazon SQS --> Queues and I can see 45 messages available.

My question: What function in the API do we call to actually get and enumerate through the messages?   For example, I called createSubscription() to create my subscription, but now what function do we call in order to enumerate through the messages?

**Comment by MeenaAmz** [comment, marked as answer]
To retrieve messages from the SQS queue programmatically, you would need to use SQS API. Please refer to this [doc](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/APIReference/API_ReceiveMessage.html) for more details. The notifications that you receive will be in JSON [format](https://github.com/amzn/selling-partner-api-models/blob/main/schemas/notifications/AnyOfferChangedNotification.json) which you can then process.
